From 4fd77a75a0dd8dbde001fe6e9e4b2bd52bc4eabb Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Tue, 28 Mar 2017 15:48:24 +0800
Subject: [PATCH 1697/5242] MLK-14530-02 driver: cpufreq: Improve the the
 cpufreq for imx7ulp

commit  e0ad38910b61926f0bb946c42dec87d93d2ba51d from
https://source.codeaurora.org/external/imx/linux-imx.git

The pm Qos add/remove only need to be called when A7 change mode
between HSRUN and RUN mode.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/cpufreq/imx7ulp-cpufreq.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/cpufreq/imx7ulp-cpufreq.c b/drivers/cpufreq/imx7ulp-cpufreq.c
index 33885cf..f2b95d7 100644
--- a/drivers/cpufreq/imx7ulp-cpufreq.c
+++ b/drivers/cpufreq/imx7ulp-cpufreq.c
@@ -108,7 +108,8 @@ static int imx7ulp_set_target(struct cpufreq_policy *policy, unsigned int index)
 		clk_set_rate(spll_pfd0, new_freq * 1000);
 		clk_set_parent(sys_sel, spll_sel);
 		clk_set_parent(arm_clk, core_div);
-		pm_qos_remove_request(&pm_qos_hsrun);
+		if (old_freq > MAX_RUN_FREQ)
+			pm_qos_remove_request(&pm_qos_hsrun);
 	}
 
 	/* scaling down? scaling voltage after frequency */
-- 
1.7.9.5

